test/core/tsi/s2a/s2a_test_util.cc

```diff
@@ -185,3 +187,220 @@ size_t expected_message_size(size_t plaintext_size) {
   return 5u /* header */ + plaintext_size + 16u /* tag */ +
          1u /* record type */;
 }
+
+bool check_encrypt_record(TLSCiphersuite ciphersuite, uint8_t* record_one,
+                          size_t record_one_size, uint8_t* record_two,
+                          size_t record_two_size, uint8_t* record_three,
+                          size_t record_three_size, char** error_details) {
+  if (record_one == nullptr && record_one_size != 0) {
+    *error_details = gpr_strdup(
+        "If |record_one| is nullptr, then |record_one_size| must be zero.");
+    return false;
+  }
+  if (record_two == nullptr && record_two_size != 0) {
+    *error_details = gpr_strdup(
+        "If |record_two| is nullptr, then |record_two_size| must be zero.");
+    return false;
+  }
+  if (record_three == nullptr && record_three_size != 0) {
+    *error_details = gpr_strdup(
+        "If |record_three| is nullptr, then |record_three_size| must be zero.");
+    return false;
+  }
+  switch (ciphersuite) {
+    case TLS_AES_128_GCM_SHA256_ciphersuite: {
+      /**  This record arises from the plaintext 123456 when encrypted using the
+       *   AES-128-GCM cipher created by the create_example_session_state
+       *   method. In decimal form, the encrypted record is the following:
+       *   {23, 3, 3, 0, 23, 242, 228, 228, 17, 172, 103, 96, 228, 227,
+       *   240, 116, 163, 101, 116, 196, 94, 228, 193, 144, 97, 3, 219, 13}.
+       */
+      uint8_t correct_record_one_bytes[28] = {
```

Comment:
Please define hard coded buffers like these as const at the top of the file.

Reply:
Also, out of curiosity, how is this generated?

Reply:
Done. These examples were generated using sslcryptor.